### PR TITLE
release notes ga - 5.24.0

### PR DESCRIPTION
**Pull request by ScottSuarez**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
 ## 5.24.0 (Unreleased)
+IMPROVEMENTS:
+* cloudrunv2: added `template.volumes.nfs` field to `google_cloud_run_v2_job` resource (beta) ([#17732](https://github.com/hashicorp/terraform-provider-google/pull/17732))
```
Comment (melinath) [inline CHANGELOG.md:3]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+* storage: added `project_number` attribute to `google_storage_bucket` resource and data source ([#17719](https://github.com/hashicorp/terraform-provider-google/pull/17719))
+* storage: added ability to provide `project` argument to `google_storage_bucket` data source. This will not impact reading the resource's data, instead this helps users avoid calls to the Compute API within the data source. ([#17719](https://github.com/hashicorp/terraform-provider-google/pull/17719))
+
+BUG FIXES:
+* appengine: fixed a crash in `google_app_engine_flexible_app_version` due to the `deployment` field not being returned by the API ([#17744](https://github.com/hashicorp/terraform-provider-google/pull/17744))
+* bigquery: fixed a crash when `google_bigquery_table` had a `primary_key.columns` entry set to `""` ([#17721](https://github.com/hashicorp/terraform-provider-google/pull/17721))
+* compute: Added explicit update_encoder to `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#17733](https://github.com/hashicorp/terraform-provider-google/pull/17733))
```
Comment (melinath) [inline CHANGELOG.md:14]: this should use the TF resource names rather than API resource names.

**Review by melinath — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+* dataflow: fixed an issue where the provider would crash when `enableStreamingEngine` is set as a `parameter` value in `google_dataflow_flex_template_job` ([#17712](https://github.com/hashicorp/terraform-provider-google/pull/17712))
```
Comment (melinath) [inline CHANGELOG.md:15]: Looks like this is beta-only

```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
+* storage: added `project_number` attribute to `google_storage_bucket` resource and data source ([#17719](https://github.com/hashicorp/terraform-provider-google/pull/17719))
+* storage: added ability to provide `project` argument to `google_storage_bucket` data source. This will not impact reading the resource's data, instead this helps users avoid calls to the Compute API within the data source. ([#17719](https://github.com/hashicorp/terraform-provider-google/pull/17719))
+
+BUG FIXES:
+* appengine: fixed a crash in `google_app_engine_flexible_app_version` due to the `deployment` field not being returned by the API ([#17744](https://github.com/hashicorp/terraform-provider-google/pull/17744))
+* bigquery: fixed a crash when `google_bigquery_table` had a `primary_key.columns` entry set to `""` ([#17721](https://github.com/hashicorp/terraform-provider-google/pull/17721))
+* compute: added explicit update_encoder to `google_compute_region_target_https_proxy` and `google_compute_target_https_proxy` resources. ([#17733](https://github.com/hashicorp/terraform-provider-google/pull/17733))
```
Comment (ScottSuarez) [inline CHANGELOG.md:13]:
```suggestion
* compute: fixed update scenarios on`google_compute_region_target_https_proxy` and `google_compute_target_https_proxy` resources. ([#17733](https://github.com/hashicorp/terraform-provider-google/pull/17733))
```

**Pull request merged**